### PR TITLE
Ticket 3903. Diabling of Harms Download Package when file conversion or dedupe is in process. 

### PR DESCRIPTION
**Pull request by Aman-Hundal**

### CODE REVIEW

File: forms-flow-web/src/components/FOI/customComponents/Records/index.js [hunk 1]
```diff
@@ -1037,6 +1036,16 @@ export const RecordsLog = ({
     }));
   }
 
+  //function to manage download for harms option
+  const enableHarmsDonwnload = () => {
+    return !recordsObj.records.every(record => record.isredactionready || (!record.isredactionready && (record.failed || isrecordtimeout(record.created_at, RECORD_PROCESSING_HRS))));
+  }
+
+  //useEffect to manage enabling and disabling of download for harms package
+  useEffect(() => {
```
Comment (nkan-aot) [inline forms-flow-web/src/components/FOI/customComponents/Records/index.js:1045]: combine with existing useeffect on line 180

File: forms-flow-web/src/components/FOI/customComponents/Records/index.js [hunk 2]
```diff
@@ -1037,6 +1036,16 @@ export const RecordsLog = ({
     }));
   }
 
+  //function to manage download for harms option
+  const enableHarmsDonwnload = () => {
+    return !recordsObj.records.every(record => record.isredactionready || (!record.isredactionready && (record.failed || isrecordtimeout(record.created_at, RECORD_PROCESSING_HRS))));
```
Comment (nkan-aot) [inline forms-flow-web/src/components/FOI/customComponents/Records/index.js:1041]: can simplify to record.isredactionready || record.failed || isrecordtimeout(record.created_at, RECORD_PROCESSING_HRS). 
Or if you want to make logic clear can use if/else

File: forms-flow-web/src/components/FOI/customComponents/Records/index.js [hunk 3]
```diff
@@ -1087,6 +1096,24 @@ export const RecordsLog = ({
                 fullWidth
               >
                 {recordsDownloadList.map((item, index) => {
+                  if (item.id === 1 && item.disabled) {
+                    return (
+                    <Tooltip title={<div style={{fontSize: "11px"}}>File conversion and deduplication in progress</div>} key={item.id}>
+                      <div>
+                        <MenuItem
+                          className="download-menu-item"
+                          key={item.id}
```
Comment (nkan-aot) [inline forms-flow-web/src/components/FOI/customComponents/Records/index.js:1105]: can you try combining this with the bottom code block? Tooltip should be able to be disabled using disableHoverListener https://mui.com/material-ui/api/tooltip/#Tooltip-prop-disableHoverListener

File: forms-flow-web/src/components/FOI/customComponents/Records/index.js (hunk 1 above)
Comment (Aman-Hundal) [reply forms-flow-web/src/components/FOI/customComponents/Records/index.js]: Adjusted. Thanks for the feedback!

File: forms-flow-web/src/components/FOI/customComponents/Records/index.js (hunk 2 above)
Comment (Aman-Hundal) [reply forms-flow-web/src/components/FOI/customComponents/Records/index.js]: Adjusted. Thanks for the feedback!

File: forms-flow-web/src/components/FOI/customComponents/Records/index.js (hunk 3 above)
Comment (Aman-Hundal) [reply forms-flow-web/src/components/FOI/customComponents/Records/index.js]: Attempted to implement this but the Menu Items became buggy and control over the tooltip did not seem to work well. Left this code as is.

**Review by nkan-aot — APPROVED** [review]

**Pull request merged**